tests_integration/__tests__/with-config-precedence.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+const runPrettier = require("../runPrettier");
+
+test("CLI overrides take precedence without --config-precedence", () => {
+  const output = runPrettier("cli/config/", ["--print-width", "1", "**/*.js"]);
+  expect(output.stdout).toMatchSnapshot();
+  expect(output.status).toEqual(0);
+});
+
+test("CLI overrides take precedence with --config-precedence cli-override", () => {
+  const output = runPrettier("cli/config/", [
+    "--print-width",
+    "1",
+    "--config-precedence",
+    "cli-override",
+    "**/*.js"
+  ]);
+  expect(output.stdout).toMatchSnapshot();
+  expect(output.status).toEqual(0);
+});
+
+test("CLI overrides take lower precedence with --config-precedence file-override", () => {
+  const output = runPrettier("cli/config/js/", [
+    "--tab-width",
+    "1",
+    "--config-precedence",
+    "file-override",
+    "**/*.js"
+  ]);
+  expect(output.stdout).toMatchSnapshot();
+  expect(output.status).toEqual(0);
+});
+
+test("CLI overrides are still applied when no config is found with --config-precedence file-override", () => {
+  const output = runPrettier("cli/config/no-config/", [
+    "--tab-width",
+    "6",
+    "--config-precedence",
+    "file-override",
+    "**/*.js"
+  ]);
+  expect(output.stdout).toMatchSnapshot();
+  expect(output.status).toEqual(0);
+});
+
+test("CLI overrides gets ignored when config exists with --config-precedence prefer-file", () => {
+  const output = runPrettier("cli/config/js/", [
+    "--print-width",
+    "1",
+    "--tab-width",
+    "1",
+    "--config-precedence",
+    "prefer-file",
+    "**/*.js"
+  ]);
+  expect(output.stdout).toMatchSnapshot();
+  expect(output.status).toEqual(0);
+});
+
+test("CLI overrides gets applied when no config exists with --config-precedence prefer-file", () => {
```

Comment:
The case is correct but maybe we should error here instead? It doesn't make sense to pass `--no-config` and `--config-precedence=prefer-file`.

Reply:
I had to do that because the jest test suite was always finding the options

```javascript
{
    semi: false
}
```

Was thinking about fixing the runner itself but that could cause other tests to behave a bit differently and would bloat this PR. 

Happy to work on this issue in another PR

Reply:
to give a bit more context

this lines:

```javascript
        if (!options) {
          return getOptions(parsedArgs);
        }
        return getOptions(dashifiedConfig);
```

options would never be falsy for this test suite.

Reply:
Maybe make another directory without a config file?

Reply:
@azz that difrectory is meant to not have a config file note this line:

```javascript
test("CLI overrides gets applied when no config exists with --config-precedence prefer-file", () => {
  const output = runPrettier("cli/config/no-config/", [

```

This directory `"cli/config/no-config/"` is the one we used for tests no having config files